Examples/Scripts/Python/seeding.py

```diff
@@ -1,266 +1,384 @@
 import acts.examples
 
 
 u = acts.UnitConstants
+SeedingAlgorithm = Enum("SeedingAlgorithm", "Default TruthSmeared TruthEstimated")
+
+TruthSeedRanges = namedtuple(
+    "TruthSeedRanges",
+    ["rho", "z", "phi", "eta", "absEta", "pt", "nHits"],
+    defaults=[(None, None)] * 7,
+)
+
+ParticleSmearingSigmas = namedtuple(
+    "ParticleSmearingSigmas",
+    ["d0", "d0PtA", "d0PtB", "z0", "z0PtA", "z0PtB", "t0", "phi", "theta", "pRel"],
+    defaults=[None] * 10,
+)
+
+SeedfinderConfigArg = namedtuple(
+    "SeedfinderConfig",
+    [
+        "maxSeedsPerSpM",
+        "cotThetaMax",
+        "sigmaScattering",
+        "radLengthPerSeed",
+        "minPt",
+        "bFieldInZ",
+        "impactMax",
+        "deltaR",  # (min,max)
+        "collisionRegion",  # (min,max)
+        "r",  # (min,max)
+        "z",  # (min,max)
+        "beamPos",  # (x,y)
+    ],
+    defaults=[None] * 7 + [(None, None)] * 5,
+)
+
+TrackParamsEstimationConfig = namedtuple(
+    "TrackParamsEstimationConfig",
+    [
+        "deltaR",  # (min,max)
+    ],
+    defaults=[(None, None)],
+)
+
+
+@acts.examples.NamedTypeArgs(
+    seedingAlgorithm=SeedingAlgorithm,
+    truthSeedRanges=TruthSeedRanges,
+    particleSmearingSigmas=ParticleSmearingSigmas,
+    seedfinderConfigArg=SeedfinderConfigArg,
+    trackParamsEstimationConfig=TrackParamsEstimationConfig,
+    logLevel=acts.logging.Level,
+)
+def addSeeding(
+    s: acts.examples.Sequencer,
+    trackingGeometry: acts.TrackingGeometry,
+    field: acts.MagneticFieldProvider,
+    geoSelectionConfigFile: Optional[Union[Path, str]] = None,
+    seedingAlgorithm: SeedingAlgorithm = SeedingAlgorithm.Default,
+    truthSeedRanges: TruthSeedRanges = TruthSeedRanges(),
+    particleSmearingSigmas: ParticleSmearingSigmas = ParticleSmearingSigmas(),
+    initialVarInflation: Optional[list] = None,
+    seedfinderConfigArg: SeedfinderConfigArg = SeedfinderConfigArg(),
+    trackParamsEstimationConfig: TrackParamsEstimationConfig = TrackParamsEstimationConfig(),
+    inputParticles="particles_initial",
+    outputDirRoot: Optional[Union[Path, str]] = None,
+    logLevel: Optional[acts.logging.Level] = None,
+    rnd: Optional[acts.examples.RandomNumbers] = None,
+) -> acts.examples.Sequencer:
+    """This function steers the seeding
+
+    Parameters
+    ----------
+    s: Sequencer
+        the sequencer module to which we add the Seeding steps (returned from addSeeding)
+    trackingGeometry : tracking geometry
+    field : magnetic field
+    geoSelectionConfigFile : Path|str, path, None
+        Json file for space point geometry selection. Not required for SeedingAlgorithm.TruthSmeared.
+    outputDirRoot : Path|str, path, None
+        the output folder for the Root output, None triggers no output
+    logLevel : acts.logging.Level, None
+        logging level to override setting given in `s`
+    rnd : RandomNumbers, None
+        random number generator. Only used by SeedingAlgorithm.TruthSmeared.
+    """
```

Comment:
- [ ] Missing `seedingAlgorithm`  description in docstring
- [ ] Missing `truthSeedRanges`  description in docstring
- [ ] Missing `particleSmearingSigmas`  description in docstring
- [ ] Missing `initialVarInflation`  description in docstring
- [ ] Missing `seedfinderConfigArg`  description in docstring
- [ ] Missing `trackParamsEstimationConfig`  description in docstring
- [ ] Missing `inputParticles`  description in docstring

Reply:
Added. Thanks for spotting this.